libraries/AP_FETtecOneWire/AP_FETtecOneWire.h

```diff
@@ -0,0 +1,253 @@
+#endif
+
+    /**
+        if init is complete sends a single fast-throttle frame containing the throttle for all found OneWire ESCs.
+        @param motor_values a 16bit array containing the throttle values that should be sent to the motors. 0-2000 where 1001-2000 is positive rotation and 0-999 reversed rotation
+        @param tlm_request the ESC to request telemetry from (-1 for no telemetry, 0 for ESC1, 1 for ESC2, 2 for ESC3, ...)
+    */
+    void escs_set_values(const uint16_t *motor_values, const int8_t tlm_request);
+
+    static constexpr uint8_t SERIAL_NR_BITWIDTH = 12;
+
+    typedef struct FETtecOneWireESC
+    {
+        bool active;
+#if HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO
+        uint8_t firmware_version;
+        uint8_t firmware_sub_version;
+        uint8_t esc_type;
+        uint8_t serial_number[SERIAL_NR_BITWIDTH];
+#endif
+    } FETtecOneWireESC_t;
+
+    FETtecOneWireESC_t _found_escs[MOTOR_COUNT_MAX]; ///< Zero-indexed array
+    uint32_t _last_config_check_ms;
+#if HAL_WITH_ESC_TELEM
+    float _crc_error_rate_factor; ///< multiply factor. Used to avoid division operations
+    uint16_t _error_count[MOTOR_COUNT_MAX]; ///< error counter from the ESCs. Zero-indexed array
+    uint16_t _error_count_since_overflow[MOTOR_COUNT_MAX]; ///< error counter from the ESCs to pass the overflow. Zero-indexed array
+    uint16_t _send_msg_count; ///< number of fast-throttle commands send by the flight controller
+    uint16_t _update_rate_hz;
+#endif
+    uint16_t _mask;
+    uint8_t _nr_escs_in_bitmask; ///< number of ESCs set on the FTW_MASK parameter
+    uint8_t _found_escs_count;   ///< number of ESCs auto-scanned in the OneWire bus by the scan_escs() function
+
+    int8_t _requested_telemetry_from_esc = -1; ///< the ESC to request telemetry from (-1 for no telemetry, 0 for ESC1, 1 for ESC2, 2 for ESC3, ...)
+    bool _initialised;       ///< device driver and ESCs are fully initialized
+    bool _uart_initialised;  ///< serial UART is fully initialized
+    bool _pull_busy;         ///< request-reply transaction is busy
+
+    enum msg_type
+    {
+        OW_OK = 0,
+        OW_BL_PAGE_CORRECT,   // BL only
+        OW_NOT_OK,
+        OW_BL_START_FW,       // BL only
+        OW_BL_PAGES_TO_FLASH, // BL only
+        OW_REQ_TYPE,
+        OW_REQ_SN,
+        OW_REQ_SW_VER,
+        OW_BEEP = 13,
+        OW_SET_FAST_COM_LENGTH = 26,
+        OW_SET_TLM_TYPE = 27, //1 for alternative telemetry. ESC sends full telem per ESC: Temp, Volt, Current, ERPM, Consumption, CrcErrCount
+        OW_SET_LED_TMP_COLOR = 51,
+    };
+
+    enum scan_state_t : uint8_t {
+        WAIT_FOR_BOOT,
+        IN_BOOTLOADER,
+        START_FW,
+        WAIT_START_FW,
+        ESC_TYPE,
+        SW_VER,
+        SN,
+        NEXT_ID,
+        CONFIG_FAST_THROTTLE,
+        CONFIG_TLM,
+        CONFIG_NEXT_ACTIVE_ESC,
+        DONE
+    };

```

Comment:
This is replaced by my suggested `enum class ESCState`

Reply:
This will forbid me to do ```_scan.state++```. To do that I would need to add an ++ operator, or explicitly write the enums.

I prefer the current solution, on this specific case.

Reply:
done by you.